### ISSUE TITLE
flaky behaviour for the .has method

### ISSUE THREAD

**Issue by elliot40404** [jaredwray/keyv#949]
**Describe the bug**
The `.has` function has unpredictable behavior when used with a `ttl` 

**How To Reproduce (best to provide workable code or tests!)**
Has been tested with node versions 16, 18 and 20 on 3 separate machines (windows, linux and mac)

Doesn't work.
```js
import Keyv from 'keyv';
import { setTimeout as sleep } from 'node:timers/promises';

const cache = new Keyv();

await cache.set('token', 1, 2000);
await sleep(1000);
console.log(await cache.has('token')); // should evaluate to true - is true
await sleep(1000);
console.log(await cache.has('token')); // should evaluate to false - is true
```

Works
```js
import Keyv from 'keyv';
import { setTimeout as sleep } from 'node:timers/promises';

const cache = new Keyv();

await cache.set('token', 1, 2000);
await sleep(1000);
console.log(await cache.get('token'), await cache.has('token')); // should evaluate to 1 true - is 1 true
await sleep(1000);
console.log(await cache.get('token'), await cache.has('token')); // should evaluate to undefined false - is undefined false
```

What i could gather from the source code is the `.has` method doesn't check for the `ttl` itself and directly used the `.get` method of the underlying storage driver, A map in my case.

Thank you very much. This library is really helpful

**Comment by jaredwray** [comment]
@elliot40404 - thanks for sending this over and I will be working to test this and if if so get a fix out with v5 of keyv which should be in alpha this quarter.

**Comment by elliot40404** [comment]
Hey @jaredwray, thanks for looking into this. I can take this on if you like and create a PR.

**Comment by jaredwray** [comment]
That would be great. Do you want to work on `main` branch for this?

**Comment by elliot40404** [comment]
Yes I can work on main, but if you have any other branch on mind please do let me know.

**Comment by jaredwray** [comment]
@elliot40404 - this was a pretty nasty bug and we added in more code to handle all the use cases. Here all all the cases we now handle on `has` and it will be released in v5 when alpha goes live in the next couple weeks. 

* When it is `Map()` the has does not handle expiring so we need to detect that type and handle it for that case. 
* If it is a Keyv Storage Adapter we need to assume they will handle the expiring correctly. Note, we found a bug in `@keyv/sqlite` and have plans to fix that also in the next 30 days where it doesnt handle the `expires`.
* If it does not have a `has` function on the storage adapter then we handle it all ourselves and check the `expires`

Nice find!
